app/model/jobs/steps/ModifyContentTags.scala

```diff
@@ -0,0 +1,70 @@
+package model.jobs.steps
+
+import com.gu.tagmanagement.{OperationType, TagWithSection, TaggingOperation}
+import model.{BatchTagOperation, Section, Tag, TagAudit}
+import model.jobs.StepStatus
+
+import scala.concurrent.duration._
+import model.jobs.Step
+import play.api.Logger
+import services.KinesisStreams
+import repositories.{ContentAPI, TagAuditRepository}
+
+import scala.language.postfixOps
+
+// The 'op' argument of this case class should really be the BatchTagOperation type but serializing enums to JSON isn't very easy with the
+// current version of play and enumeratim
+case class ModifyContentTags(tag: Tag, section: Option[Section], contentIds: List[String], op: String, `type`: String = ModifyContentTags.`type`, var stepStatus: String = StepStatus.ready, var stepMessage: String = "Waiting", var attempts: Int = 0) extends Step {
+  override def process = {
+    contentIds foreach { contentPath =>
+      val taggingOperation = TaggingOperation(
+        operation = OperationType.valueOf(op.replace("-", "")).get,
+        contentPath = contentPath,
+        tag = Some(TagWithSection(tag.asThrift, section.map(_.asThrift)))
+      )
+      KinesisStreams.taggingOperationsStream.publishUpdate(contentPath.take(128), taggingOperation)
```

Comment:
Why do we take `128` here? Could pull out in a val for readability to explain this.

Reply:
The online docs say that the maximum length of a kinesis partition key is 256 bytes and JVM strings are fixed 16 bytes wide so we limit the partition key length to 128 characters.

Weirdly enough the java doc contradicts this and states the maximum length is 256 "unicode characters" whatever that means.